internal/executor/executor.go

```diff
@@ -151,14 +156,36 @@ func (executor *Executor) RunBuild(ctx context.Context) {
 		}
 
 		log.Printf("Executing %s...", command.Name)
-		if err := executor.performStep(subCtx, environment, command); err != nil {
-			if errors.Is(err, ErrStepExit) {
-				return
-			}
 
+		stepResult, err := executor.performStep(subCtx, environment, command)
+		if err != nil {
+			return
+		}
+
+		if !stepResult.Success {
 			failedAtLeastOnce = true
 		}
+
 		log.Printf("%s finished!", command.Name)
+
+		_ = retry.Do(
+			func() error {
+				_, err := client.CirrusClient.ReportSingleCommand(ctx, &api.ReportSingleCommandRequest{
+					TaskIdentification: executor.taskIdentification,
+					CommandName:        command.Name,
+					Succeded:           stepResult.Success,
+					DurationInSeconds:  int64(stepResult.Duration.Seconds()),
+					SignaledToExit:     stepResult.SignaledToExit,
+					LocalTimestamp:     time.Now().Unix(),
+				})
+				return err
+			}, retry.OnRetry(func(n uint, err error) {
+				log.Printf("Failed to report command %v: %v\nRetrying...\n", command.Name, err)
+			}),
+			retry.Delay(10*time.Second),
+			retry.Attempts(2),
+			retry.Context(ctx),
+		)
```

Comment:
We should probably remove it once #41 is implemented. Only retry the new stats method which will finalize the task execution.